src/Diagnostics.RuntimeHost/Controllers/HostingEnvironment/HostingEnvironmentController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Diagnostics.ModelsAndUtils.Models;
+using Diagnostics.RuntimeHost.Models;
+using Diagnostics.RuntimeHost.Services;
+using Diagnostics.RuntimeHost.Services.SourceWatcher;
+using Diagnostics.RuntimeHost.Utilities;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Diagnostics.RuntimeHost.Controllers
+{
+    [Produces("application/json")]
+    [Route(UriElements.HostingEnvironmentResource + UriElements.Diagnostics)]
+    public sealed class HostingEnvironmentController : HostingEnvironmentControllerBase
+    {
+        public HostingEnvironmentController(IStampService stampService, ICompilerHostClient compilerHostClient, ISourceWatcherService sourceWatcherService, IInvokerCacheService invokerCache, IDataSourcesConfigurationService dataSourcesConfigService)
+            : base(stampService, compilerHostClient, sourceWatcherService, invokerCache, dataSourcesConfigService)
+        {
+        }
+
+        [HttpPost(UriElements.Query)]
+        public async Task<IActionResult> Post(string subscriptionId, string resourceGroupName, string hostingEnvironmentName, string[] hostNames, string stampName, [FromBody]CompilationBostBody<DiagnosticStampData> jsonBody, string startTime = null, string endTime = null, string timeGrain = null)
```

Comment:
CompilationBostBody supposed to be CompilationPostBody?

Reply:
lol .. will fix

Reply:
Aspaan comes back to haunt us!